pilot/cmd/pilot-agent/main.go

```diff
@@ -434,13 +428,17 @@ var (
 				ControlPlaneAuth:    controlPlaneAuthEnabled,
 				DisableReportCalls:  disableInternalTelemetry,
 			})
-			agent := envoy.NewAgent(envoyProxy, envoy.DefaultRetry, features.TerminationDrainDuration())
+
+			agent := envoy.NewAgent(envoyProxy, features.TerminationDrainDuration())
+
 			watcher := envoy.NewWatcher(tlsCertsToWatch, agent.ConfigCh())
 
-			go waitForCompletion(ctx, agent.Run)
-			go waitForCompletion(ctx, watcher.Run)
+			go watcher.Run(ctx)
+
+			// On SIGINT or SIGTERM, cancel the context, triggering a graceful shutdown
+			go cmd.WaitSignalFunc(cancel)
 
-			cmd.WaitSignal(make(chan struct{}))
+			agent.Run(ctx)
```

Comment:
I feel like we should have `Run` return the error from envoy. Not sure how the error is used, however ... maybe not terribly important.

Reply:
I agree. It should return an `error`